main/autobot.js

```diff
@@ -9,9 +9,8 @@ class Autobot {
     this.handler = null;
   }
 
-  receive(input, cb) {
-    this.handler = new Handler(cb);
-    this.adapter.receive(input);
+  receive(input) {
```

Comment:
Let's keep the callback here, and then have 
`this.adapter.receive(input).then(callback)`